src/custom/state/cowToken/hooks.ts

```diff
@@ -166,16 +168,26 @@ export function useCowBalance() {
  * Hook that returns combined vCOW + COW balance + vCow from locked GNO
  */
 export function useCombinedBalance() {
+  const { chainId, account } = useActiveWeb3React()
   const { total: vCowBalance } = useVCowData()
   const lockedGnoBalance = useAllocation()
   const cowBalance = useCowBalance()
 
   return useMemo(() => {
-    if (!vCowBalance || !cowBalance || !lockedGnoBalance) {
-      return
+    let tmpBalance = JSBI.BigInt(0)
+
+    const isLoading = account && (!vCowBalance || !lockedGnoBalance || !cowBalance) ? true : false
+
+    const cow = COW[chainId || SupportedChainId.MAINNET]
+
+    if (account) {
+      if (vCowBalance) tmpBalance = JSBI.add(tmpBalance, vCowBalance.quotient)
+      if (lockedGnoBalance) tmpBalance = JSBI.add(tmpBalance, lockedGnoBalance.quotient)
```

Comment:
I think this is not quite correct: We're adding the entire allocation without taking into account how much of it has already been claimed by the user. That share of the allocation that has been claimed is already in the user's wallet as COW and thus is counted towards the fee discount.

We should use the `useCowFromLockedGnoBalances` hook instead and then only add `allocated - claimed` here.

Reply:
@jfschwarz Good point, I've added fix for this in this PR https://github.com/cowprotocol/cowswap/pull/460 (in case we want to add this to fee discount).